php-checks/src/main/java/org/sonar/php/checks/HardCodedCredentialsInVariablesAndUrisCheck.java

```diff
@@ -180,10 +104,13 @@ private void checkForCredentialUri(LiteralTree literal) {
 
     if (uri.getUserInfo() != null) {
       String userInfo = uri.getUserInfo();
-      Matcher m = Pattern.compile("(\\S+):(\\S+)").matcher(userInfo);
-      if (m.find() && !m.group(1).equals(m.group(2)) && !userInfo.equals("user:password") && !userInfo.equals("username:password")) {
-        context().newIssue(this, literal, MESSAGE_URI);
+
+      String[] splitUserInfo = userInfo.split(":");
+      if (splitUserInfo.length == 1 || splitUserInfo[0].equals(splitUserInfo[1]) || "user:password".equals(userInfo)
+        || ("username:password").equals(userInfo)) {
+        return;
       }
```

Comment:
Nice improvement. It's much more performant and easier to read. You could move `user:password".equals(userInfo) || ("username:password").equals(userInfo)` into a static method to make the condition more readable. Like `isCommonTestCredential`